controllers/machine_controller.go

```diff
@@ -83,6 +89,47 @@ func (r *MachineReconciler) SetupWithManager(mgr ctrl.Manager, options controlle
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
 
+	mapFn := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			return []reconcile.Request{
+				{NamespacedName: types.NamespacedName{
+					Name:      a.Meta.GetName(),
+					Namespace: a.Meta.GetNamespace(),
+				}},
+			}
+		})
+
+	// 'UpdateFunc' and 'CreateFunc' used to judge if a event about the object is
+	// what we want. If that is true, the event will be processed by the reconciler.
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldCluster := e.ObjectOld.(*clusterv1.Cluster)
+			newCluster := e.ObjectNew.(*clusterv1.Cluster)
+			if oldCluster.Spec.Paused && !newCluster.Spec.Paused {
```

Comment:
ooh, I like the use of predicates here...  I wonder if we can make use of these in other places to limit when we are queuing resources for reconciliation :)